---
title: "06_annotate_mags.md"
author: "Gulnara Tagridzhanova"
date: "28/08/2023"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "/Users/gol22pin/Documents/Lichen_project_TSL/01_Xanthoria_metagenomics/notebook")

```

**Rationale:** need to annotate all MAGs to use them for metatranscriptomic analysis

## 1. Preparing MAGs
* Implemented the whole thing as a snakemake pipeline
* Created new folders to host annotations
```
mkdir analysis_and_temp_files/06_annotate_mags/euk
mkdir analysis_and_temp_files/06_annotate_mags/prok
```
* For both eukaryotic and prokaryotic MAGs, first cleaned, sorted, and renames fastas using funannotate


## 2. Bacterial MAGs
* Install fresh prokka to fix the problem with tbl2asn expiry
```
ssh software
cd /tsl/scratch/gol22pin/singularity
singularity build prokka.sif docker://staphb/prokka:latest
```
* Annotate as a part of the snakemake pipeline

## 3. Eukaryotic MAGs
### Repeat-masking
* Used RpeatMasker and RepBase
  * For fungal MAGs used fngrep.ref, which has annotated repeats from across fungi
  * For algal MAGs used chlrep.ref, whihc has annotated repeats from Chlamydomonas
* Test run on one fungal and one algal MAG
  * Fungal: masked only 0.4%, including both simple repeats and unclassified interspersed repeats
  * Algal: masked 2%, also both simple repeats and unclassified interspersed repeats
  
```
source package /tsl/software/testing/bin/repeatmasker-4.0.9 
RepeatMasker -pa 5 -a -s -gff -xsmall -lib /tsl/software/testing/repeatmasker/RepeatMasker/4.0.3/x86_64/bin/Libraries/RepBase18.08.fasta/fngrep.ref  analysis_and_temp_files/06_annotate_mags/euk/GTX0466.bin.15_cleaned_sorted.fa &> analysis_and_temp_files/06_annotate_mags/euk/GTX0466.bin.15_cleaned_sorted.repeatmasker.out 
==================================================
file name: GTX0466.bin.15_cleaned_sorted.fa
sequences:           110
total length:   27330910 bp  (27330910 bp excl N/X-runs)
GC level:         51.25 %
bases masked:     115475 bp ( 0.42 %)
==================================================
               number of      length   percentage
               elements*    occupied  of sequence
--------------------------------------------------
SINEs:                0            0 bp    0.00 %
      ALUs            0            0 bp    0.00 %
      MIRs            0            0 bp    0.00 %

LINEs:                0            0 bp    0.00 %
      LINE1           0            0 bp    0.00 %
      LINE2           0            0 bp    0.00 %
      L3/CR1          0            0 bp    0.00 %

LTR elements:         0            0 bp    0.00 %
      ERVL            0            0 bp    0.00 %
      ERVL-MaLRs      0            0 bp    0.00 %
      ERV_classI      0            0 bp    0.00 %
      ERV_classII     0            0 bp    0.00 %

DNA elements:         0            0 bp    0.00 %
     hAT-Charlie      0            0 bp    0.00 %
     TcMar-Tigger     0            0 bp    0.00 %

Unclassified:       261        26612 bp    0.10 %

Total interspersed repeats:    26612 bp    0.10 %


Small RNA:            0            0 bp    0.00 %

Satellites:           0            0 bp    0.00 %
Simple repeats:    2102        91025 bp    0.33 %
Low complexity:       0            0 bp    0.00 %
==================================================


source package /tsl/software/testing/bin/repeatmasker-4.0.9 
RepeatMasker -pa 5 -a -s -gff -xsmall -lib /tsl/software/testing/repeatmasker/RepeatMasker/4.0.3/x86_64/bin/Libraries/RepBase18.08.fasta/chlrep.ref  analysis_and_temp_files/06_annotate_mags/euk/GTX0465.bin.1_cleaned_sorted.fa &> analysis_and_temp_files/06_annotate_mags/euk/GTX0465.bin.1_cleaned_sorted.repeatmasker.out 
==================================================
file name: GTX0465.bin.1_cleaned_sorted.fa
sequences:          1796
total length:   59369835 bp  (59369835 bp excl N/X-runs)
GC level:         49.62 %
bases masked:    1167210 bp ( 1.97 %)
==================================================
               number of      length   percentage
               elements*    occupied  of sequence
--------------------------------------------------
SINEs:                0            0 bp    0.00 %
      ALUs            0            0 bp    0.00 %
      MIRs            0            0 bp    0.00 %

LINEs:                0            0 bp    0.00 %
      LINE1           0            0 bp    0.00 %
      LINE2           0            0 bp    0.00 %
      L3/CR1          0            0 bp    0.00 %

LTR elements:         0            0 bp    0.00 %
      ERVL            0            0 bp    0.00 %
      ERVL-MaLRs      0            0 bp    0.00 %
      ERV_classI      0            0 bp    0.00 %
      ERV_classII     0            0 bp    0.00 %

DNA elements:         0            0 bp    0.00 %
     hAT-Charlie      0            0 bp    0.00 %
     TcMar-Tigger     0            0 bp    0.00 %

Unclassified:      1209       126600 bp    0.21 %

Total interspersed repeats:   126600 bp    0.21 %


Small RNA:            0            0 bp    0.00 %

Satellites:           0            0 bp    0.00 %
Simple repeats:   21134      1077629 bp    1.82 %
Low complexity:       0            0 bp    0.00 %
==================================================
```
  
### GeneMark
* Had to run GeneMark outside of funannotate, because according to the author GeneMark cannot be reached by funannotate from its container
* Used a GM copy from HPC
```
bash code/annotate_funannotate.sh analysis_and_temp_files/06_annotate_mags/euk/coassembly.bin.195_cleaned_sorted.fa.masked analysis_and_temp_files/06_annotate_mags/euk/coassembly.bin.195_pred/predict_results/coassembly.bin.195.proteins.fa coassemb195 10 alga

#this command was run: gmes_petap.pl --ES --max_intron 3000 --soft_mask 2000 --cores $thread --sequence ../../$filename
```

### Funannotate predict
* For algae, tried replacing BUSCO dikarya set with embryophyta, since chlorophyta aren't available 
  * This resulted in an error, as busco didn't produce enough models to train augustus
* Tried installing BUSCO db, but for that to work, I needed to re-install funannotate with a writable sandbox
* Copied chlorophyta_odb10 into the folder in container with all databases (`/opt/databases`)

* NB: Even though the environmental variables are set in the container, this isn't transferred while run outside. Need to pass environmental variable in the same command as calling singularity (with SINGULARITYENV_). 
```
#on ssh node
sudo singularity build --sandbox funannotate2 docker://nextgenusfs/funannotate 
singularity shell --writable funannotate2
funannotate setup -b eukaryota
funannotate setup -i all
#on regular node
cd ../01_Xanthoria_metagenomics
SINGULARITYENV_PATH=/venv/bin/:$PATH singularity exec ../singularity/funannotate2 funannotate check --show-versions

```
* Test run worked
```
bash code/annotate_funannotate.sh analysis_and_temp_files/06_annotate_mags/euk/coassembly.bin.195_cleaned_sorted.fa.masked analysis_and_temp_files/06_annotate_mags/euk/coassembly.bin.195_pred/predict_results/coassembly.bin.195.proteins.fa coassemb195 10 alga coassembly.bin.195

SINGULARITYENV_PATH=/venv/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin SINGULARITYENV_AUGUSTUS_CONFIG_PATH=/usr/share/augustus/config SINGULARITYENV_EVM_HOME=/venv/opt/evidencemodeler-1.1.1 singularity exec ../singularity/funannotate2 funannotate predict -i $assemblymasked -o $outdir --optimize_augustus --cpus $thread --name $prefix -s $prefix --organism $organism --busco_db $busco_lineage --genemark_gtf $outdir/genemark/genemark.gtf -d /opt/databases --busco_seed_species $busco_seed_species
```

* Actual run: in 4 assemblies out of 11, GeneMark gave an error:
  * coassembly.bin.376 (low coverage, non-mycobiont Lecanoromycete): "error, not enough sequnce to run training in file"
  * coassembly.bin.378 (low coverage, non-mycobiont Lecanoromycete): "error, not enough sequnce to run training in file"
  * coassembly.bin.76 (Chaetothyriales): "error, not enough sequnce to run training in file"
  * coassembly.bin.64 (Chaetothyriales): "error, file not found: data/training.fna"
* [Acording to the funannotate creators](https://github.com/nextgenusfs/funannotate/issues/287), this might be due to too fragmented assembly
* All of these genomes are very fragmented, with mean scaffold <10,000 bp
* All of them from the co-assembly
* Two of them (coassembly.bin.376 and coassembly.bin.64) are recovered only in coassembly, so there is no alternatives
* The other two are also recovered in individual metagenomes, but the quality of the assembly isn't better there
* **Solution:** since I can't produce better assemblies, I'll skip genemark entirely. Re-ran the script with genemark part commented

 
