---
title: "04_characterize_secretome.md"
author: "Gulnara Tagridzhanova"
date: "30/01/2024"
output: 
  html_document:
    code_folding: hide
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "/Users/gol22pin/Documents/Lichen_project_TSL/10_lichen_effectors/notebook")
library(kableExtra)
library(stringr)
library(tidyverse)
```
**Summary:** characterize secretome based on funannotate and matches against PDB

## 1. Structural similarity to PDB (FoldSeek search against PDB)
* Angus ran search against PBD on the HPC
* Download date 11.12.2023
```{r,eval=F}
#!/bin/bash -e

#SBATCH -p tsl-medium
#SBATCH --mem=20G
#SBATCH -c 4
#SBATCH -J foldseek
#SBATCH --mail-type=begin,end,fail
#SBATCH --mail-user=angus.bucknell@tsl.ac.uk
#SBATCH -o slurm.%j.out
#SBATCH -e slurm.%j.err


source package cda29b6a-320e-4d73-83c6-240ed7a6201e
foldseek easy-search lichen-effectors ../pdb_db/pdb output tmp --format-output "query,target,nident,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,prob,evalue,bits" --max-seqs 5
```

* Read the results
```{r}
df<-read.delim2("../analysis_and_temp_files/04_characterize_secretome/output.txt",header=F)
colnames(df)<-c("query","target","nident","fident","alnlen","mismatch","gapopen","qstart","qend","tstart","tend","prob","evalue","bits")
df$TranscriptID<- str_match(df$query, "_(.*?)unrelaxed")[,2]
df$TranscriptID<-str_replace(df$TranscriptID,"T","-T") 
df$TranscriptID<-str_replace(df$TranscriptID,"FUN","XANPAGTX0501_")  
df$evalue<-as.numeric(df$evalue)
```
* Of the 608 secreted proteins, `r df$TranscriptID %>% unique %>% length()` have at least one hit (regardless of the score and evalue)
* `r df$TranscriptID[df$evalue<0.001] %>% unique %>% length()` have hits with e-value < 0.001 (I took the cut-off value from the [FoldSeek paper](https://www.nature.com/articles/s41586-023-06510-w.pdf) )
* Connect PDB IDs to descriptions
  * For that, I downloaded the index file from the current version of PDB (accessed on 30.01.24) `https://files.rcsb.org/pub/pdb/derived_data/index/compound.idx`

```{r}
pdb_index<-read.delim2("../analysis_and_temp_files/04_characterize_secretome/compound.idx",header=F,skip=4)
colnames(pdb_index)<-c("pdb_id","pdb_description")

df$pdb_id<-gsub("\\..*", "", df$target) %>% toupper()
df<-df %>% left_join(pdb_index)
```

* For each protein, select best three matches (with e-value below 0.001)
```{r}
best<-df %>% filter(evalue<0.001) %>% group_by(TranscriptID) %>%
    arrange(TranscriptID, evalue) %>% 
    mutate(indx = row_number()) %>%
  filter(indx<=3) %>%
  mutate(match = case_when(indx==1 ~ "pdb_best",
                           indx==2 ~ "pdb_second",
                           indx==3 ~ "pdb_third"),
         pdb=paste(pdb_id,pdb_description,evalue,sep="; ")) %>%
           select(TranscriptID,pdb,match) %>%
  pivot_wider(names_from = "match",values_from = pdb)
```

## 2. Antimicrobial activity with Amapec
* Used [Amapec](https://github.com/fantin-mesny/amapec) v1.0b to predict which proteins have antimicrobial activity
* Used the colab notebook provided by the developers. As input, used ColabFold pdb files
* The developers say that bad models (with low pLDDT) can be erroneously assigned as antimicrobial. To avoid that, I used a cut-off of pLDDT>=65, following the cut-off that Adam used in FoldTree
* Got 128 proteins predicted as antimicrobial
```{r}
amapec<-read.delim2("../analysis_and_temp_files/04_characterize_secretome/amapec.csv",sep=",")
amapec$TranscriptID<- str_match(amapec$Protein.ID, "(.*?)_unrelaxed")[,2]
amapec$TranscriptID<-str_replace(amapec$TranscriptID,"FUN","XANPAGTX0501")

amapec<-amapec %>% mutate(antimicrobial = ifelse(pLDDT>=65,Prediction,"Below pLDDT threshold")) %>% select(TranscriptID,pLDDT,antimicrobial,Probability.of.antimicrobial.activity)
table(amapec$antimicrobial)
```


## 3. Structural similarity to AlphaFold 
* Angus ran search against the AlphaFold db on the HPC using the same script as above
* The database download date 18.04.2024
* Process it as above
```{r}
df_af<-read.delim2("../analysis_and_temp_files/04_characterize_secretome/lichenAFDB_tidy.csv",header=T,sep=",")
df_af$TranscriptID<- str_match(df_af$query, "_(.*?)unrelaxed")[,2]
df_af$TranscriptID<-str_replace(df_af$TranscriptID,"T","-T") 
df_af$TranscriptID<-str_replace(df_af$TranscriptID,"FUN","XANPAGTX0501_")  
df_af$evalue<-as.numeric(df_af$evalue)
```
* Of the 608 secreted proteins, `r df_af$TranscriptID %>% unique %>% length()` have at least one hit (regardless of the score and evalue)
* `r df_af$TranscriptID[df$evalue<0.001] %>% unique %>% length()` have hits with e-value < 0.001 (I took the cut-off value from the [FoldSeek paper](https://www.nature.com/articles/s41586-023-06510-w.pdf) )
* Connect AlphaFold IDs to descriptions
  * Save hit IDs as a separate file
```{r}
df_af$target_clipped<- str_match(df_af$target, "AF-(.*?)-F")[,2]
write.table(df_af$target_clipped %>% data.frame, file="../analysis_and_temp_files/04_characterize_secretome/uniprotID.txt", quote=F, sep='\t', row.names=F,col.names = F)
```
  * Retrieved UniProt info using the [tool from their website](https://www.uniprot.org/id-mapping). The tool only outputs each protein one time, so will need to merge
```{r}
uniprot<-read.delim2("../analysis_and_temp_files/04_characterize_secretome/idmapping_2024_04_23.tsv",header=T)
df_af<-df_af %>% left_join(uniprot, by=c("target_clipped"="Entry"))
df_af$prot_description<-paste0(df_af$Protein.names," from ",df_af$Organism)
```

* For each protein, select best three matches (with e-value below 0.001)
```{r}
best_af<-df_af %>% filter(evalue<0.001) %>% group_by(TranscriptID) %>%
    arrange(TranscriptID, evalue) %>% 
    mutate(indx = row_number()) %>%
  filter(indx<=3) %>%
  mutate(match = case_when(indx==1 ~ "af_best",
                           indx==2 ~ "af_second",
                           indx==3 ~ "af_third"),
         af=paste(target_clipped,Protein.names,Organism,evalue,sep="; ")) %>%
           select(TranscriptID,af,match) %>%
  pivot_wider(names_from = "match",values_from = af)
```

## 4. Add Funannotate annotations and expression in lichen/culture
* Combined together best matches in PDB, antimicrobial predictions, and functional annotations based on sequence similarities
* Added a column to indicate whether this protein upregulated in lichen or culture
* Added a column to indicate whether this protein had been examined before among the 91 outative effectors
```{r}
funannot2<-read.delim2("../../02_long_read_assemblies/analysis_and_temp_files/09_ortho/lichen_enriched_ortho_in_xanpa.tsv",sep="\t")
eff_list<-read.delim2("../analysis_and_temp_files/02_characterize_predicted/annotation_predicted_effectors.txt")
eff_list$TranscriptID<-str_replace(eff_list$TranscriptID,"FUN","XANPAGTX0501")
upp_lich<-read.delim2("../../03_transcriptomic_analysis/analysis_and_temp_files/08_dge_culture_lichen/upreg_in_lichen_sleuth.txt",sep="\t")
upp_cul<-read.delim2("../../03_transcriptomic_analysis/analysis_and_temp_files/08_dge_culture_lichen/upreg_in_culture_sleuth.txt",sep="\t")

combined <- funannot2 %>% filter(secreted_consensus==T) %>% 
  left_join(best) %>% left_join(best_af) %>% left_join(amapec) %>% 
  select(TranscriptID,pLDDT,pdb_best,pdb_second,pdb_third,af_best,af_second,af_third,InterPro_new,InterPro,antimicrobial,Probability.of.antimicrobial.activity,
         CAZyme_new,Protease_new,KO,lichen_ortho) %>%
  mutate(included_as_eff = ifelse(TranscriptID %in% eff_list$TranscriptID,T,F),
         DGE = case_when(TranscriptID %in% upp_lich$target_id ~ "Upregulated in lichen",
                         TranscriptID %in% upp_cul$target_id ~ "Upregulated in culture",
                         T ~ "Non-DGE"))
write.table(combined, file="../analysis_and_temp_files/04_characterize_secretome/secretome_annotations.txt", quote=F, sep='\t', row.names=F)

combined %>%  
  kable(format = "html", col.names = colnames(combined)) %>%
  kable_styling() %>%
  kableExtra::scroll_box(width = "100%", height = "600px")
```


* Of the 608, `r combined %>% filter(lichen_ortho==T) %>% nrow()` are from lichen-enriched orthogroups
* `r combined %>% filter(is.na(InterPro_new),is.na(CAZyme_new),is.na(Protease_new),KO=="") %>% nrow()` had no functional annotations from funannotate (this is `r (combined %>% filter(is.na(InterPro_new),is.na(CAZyme_new),is.na(Protease_new),KO=="") %>% nrow()) / nrow(combined)`; compare to `r (funannot2 %>% filter(is.na(InterPro_new),is.na(CAZyme_new),is.na(Protease_new),KO=="") %>% nrow()) / nrow(funannot2)` across the whole genome)
* Secretome had genes upregulated in lichen and in culture
```{r}
combined$DGE %>% table()
```

* `r combined %>% filter(is.na(InterPro_new),is.na(CAZyme_new),is.na(Protease_new),KO=="",!is.na(pdb_best)) %>% nrow()` proteins had no funannotate annotations, but have at least one match to PDB

```{r}
combined %>%  filter(is.na(InterPro_new),is.na(CAZyme_new),is.na(Protease_new),KO=="",!is.na(pdb_best)) %>%
  kable(format = "html", col.names = colnames(combined)) %>%
  kable_styling() %>%
  kableExtra::scroll_box(width = "100%", height = "600px")
```


* The proteins predicted as antimicrobial, mostly have PDB annotations (`r combined %>% filter(antimicrobial=="Antimicrobial",!is.na(pdb_best)) %>% nrow()` our of `r combined %>% filter(antimicrobial=="Antimicrobial") %>% nrow()`)
  * Of them, `r combined %>% filter(antimicrobial=="Antimicrobial",!is.na(pdb_best),!is.na(CAZyme_new)) %>% nrow()` are CAZymes and `r combined %>% filter(antimicrobial=="Antimicrobial",!is.na(pdb_best),!is.na(Protease_new)) %>% nrow()` are proteases
  * For comparison, across the entire secretome, only `r (combined %>% filter(!is.na(pdb_best)) %>% nrow())/(combined %>% nrow())` had PDB matches

## 4. Notable proteins
#### XANPAGTX0501_009887-T1: match to an effector from Trichoderma
* Great model and great match to 7CWJ, Tsp1, a small secreted cysteine-rich protein from T. virens
* Is strongly upregulated in lichens (b = -6.15)

>Structure and evolutionary analysis suggested that Tsp1 belongs to a novel effector protein family. Tsp1 acted as an inducer of salicylic acid mediated susceptibility in plants, rendering maize plants more susceptible to a necrotrophic pathogen Cochliobolus heterostrophus

#### XANPAGTX0501_003600-T1: killer toxin
* Great model and great match to 1KPT, a toxin inhibiting Ca2+ channel

>The P4 strain of the corn smut fungus, Ustilago maydis, secretes a fungal toxin, KP4, encoded by a fungal virus (UMV4) that persistently infects its cells. UMV4, unlike most other (non-fungal) viruses, does not spread to uninfected cells by release into the extracellular milieu during its normal life cycle and is thus dependent upon host survival for replication. In symbiosis with the host fungus, UMV4 encodes KP4 to kill other competitive strains of U. maydis, thereby promoting both host and virus survival.

#### XANPAGTX0501_005792-T1: bad match to an effector
* Okay-ish model 
* FoldSeek reports a match to a known effector 4B9H, a Cladosporium fulvum LysM effector Ecp6, but ChimeraX doesn't align them
* InterPro annotated a LysM domain. COuld it be that the match is between the LysMs?

## 5. Conclusions
* In the big list (all secretome, n = 608),many proteins from the same functional groups as proteins from the smaller list reviewed previously (putative effectors, n = 91, see `../02_characterize_selected`): GHs and proteases, lectins, thaumatin-like proteins, protease inhibitors, toxins, etc. **To me, this confirms that we should deal with the whole secretome, as our ability to select effectors bioinformatically seems to be very limited**
* FoldSeek search agains PDB  seem to miss some matches. Anecdotally, they didn't retrive a macth to our hydrophobin (XANPAGTX0501_002409-T1). In FoldSeek, we got three vary poor matches (evalue >1). When I previouslt searched PDB manually with DALLI, I got a match to hydrophobin. Could it be an issue with the database search? 
* In contrast, our search against the AlphaFOld database identified the hydrophobin correctly. However, overall this search didn't add much information compared to PDB+funannotate

